### PR TITLE
Add maxlength option to activeTextInput

### PR DESCRIPTION
**Pull request by ghost**
Issue [#7440](https://github.com/yiisoft/yii2/issues/7440).

### CODE REVIEW

File: framework/helpers/BaseHtml.php [hunk 1]
```diff
@@ -1157,6 +1157,18 @@ public static function activeInput($type, $model, $attribute, $options = [])
      */
     public static function activeTextInput($model, $attribute, $options = [])
     {
+        if (!isset($options['maxlength'])) {
```
Comment (samdark) [inline framework/helpers/BaseHtml.php]: Doesn't look right to me. Option isn't meant to be set by user, correct? Either the check should be `isset` without `!` or the whole block should be removed.

File: framework/helpers/BaseHtml.php [hunk 2]
```diff
@@ -1157,6 +1157,18 @@ public static function activeInput($type, $model, $attribute, $options = [])
...
+            return static::activeInput('text', $model, $attribute, $options);
+        }
+
+        $attrName = preg_replace('/[\[].*[\]]/U', '', $attribute);
```
Comment (cebe) [inline framework/helpers/BaseHtml.php]: this should use `Html::getAttributeName()`.

File: framework/helpers/BaseHtml.php (hunk 1 above)
Comment (cebe) [inline framework/helpers/BaseHtml.php]: better use array_key_exists to allow explicitly specifying null to not have this attribute rendered.

**Comment by cebe** [comment]
In general I am not sure if it is a good idea to enable it by default. Consider the case where I want to paste some stringe, e.g. a bank accound number into a field which is separated by spaces `DE 12 3456 789` The field has a maximum length validator because there is a max length for IBAN but I'd want to adjust the field after I pasted the text. It would not be possible to paste the string when maxlength is set.

File: framework/helpers/BaseHtml.php (hunk 1 above)
Comment (llfm) [inline framework/helpers/BaseHtml.php]: @samdark I assumed that the user would never manual type any value different than the string max value.
@cebe If the option isn't created, the input is normally rendered. Any value with the key `maxlength` will try to the find the `max` (perhaps make here a explicity value (e.g. `true`))

**Comment by llfm** [comment]
> In general I am not sure if it is a good idea to enable it by default

It isn't enabled by default.

It will only render the maxlength attribute if the key `maxlength` exists.
```php
if (!isset($options['maxlength']))
  //normal

//tries to find max
```

**Comment by cebe** [comment]
okay, then this is confusing as it will override the value I specify in maxlength value with a value from the validator.

**Comment by llfm** [comment]
Can be confusing and dictatorial, but I can't imagine right now a situation when you want to have a different `maxlength` from the `max` in the string rule.

Something like this? (true can be "auto" or something like this)
```php
if (!isset($options['maxlength']) || $options['maxlength'] !== true) {
```

**Comment by qiangxue** [comment]
I agree with @cebe's concerns, too.

**Comment by llfm** [comment]
This seems okay? (Perhaps the `true` should be a string `'auto'`)

**Comment by qiangxue** [comment]
A cleaner way is as follows:

```php
if (isset($options['maxlength']) && $options['maxlength'] === true) {
    ...
} else {
   return parent::activeTextInput(...);
}
```

This is because it's possible `$options['maxlength'] === null`

**Comment by llfm** [comment]
Makes sense. Updated.

**Comment by qiangxue** [comment]
Fixed. Thanks.

**Comment by lynicidn** [comment]
if i have many rules it's not best way :(

**Comment by creocoder** [comment]
:+1: Very DRY

**Pull request closed without merging**